src/tokens/tokens.service.ts

```diff
@@ -172,21 +191,74 @@ export class TokensService {
       },
       ...basicAuth(this.username, this.password),
     };
+    return requestOptions;
+  }
 
+  private queryOptions() {
+    const requestOptions: AxiosRequestConfig = {
+      ...basicAuth(this.username, this.password),
+    };
     return requestOptions;
   }
 
-  createPool(dto: TokenPool): TokenPoolEvent {
-    const poolId = new URLSearchParams({
+  private async queryPool(encodedPoolId: URLSearchParams) {
+    const validPoolId: ITokenPool = this.validatePoolId(encodedPoolId);
+    const nameResponse = await lastValueFrom(
+      this.http.post<EthConnectReturn>(
+        `${this.baseUrl}`,
+        {
+          headers: {
+            type: queryHeader,
+          },
+          to: validPoolId.address,
+          method: this.getMethodAbi(encodedPoolId, 'NAME'),
+          params: [],
+        } as EthConnectMsgRequest,
+        this.queryOptions(),
+      ),
+    );
+    const symbolResponse = await lastValueFrom(
+      this.http.post<EthConnectReturn>(
+        `${this.baseUrl}`,
+        {
+          headers: {
+            type: queryHeader,
+          },
+          to: validPoolId.address,
+          method: this.getMethodAbi(encodedPoolId, 'SYMBOL'),
+          params: [],
+        } as EthConnectMsgRequest,
+        this.queryOptions(),
+      ),
+    );
+    return {
+      name: nameResponse.data.output,
+      symbol: symbolResponse.data.output,
+    };
+  }
+
+  async createPool(dto: TokenPool): Promise<TokenPoolEvent> {
+    const encodedPoolId = new URLSearchParams({
       address: dto.config.address,
       standard: dto.type === TokenType.FUNGIBLE ? 'ERC20WithData' : 'ERC721WithData',
       type: dto.type,
     });
 
+    const nameAndSymbol = await this.queryPool(encodedPoolId);
+    if (nameAndSymbol.name !== dto.name) {
```

Comment:
Should it be mandatory to specify the name in the case of an existing ERC20/ERC721, or should we allow you to just specify the address and we populate the other fields?

... feels a little arbitrary to force you to specify it, then require it to be identical to the value we queried.

I understand there might be a cost vs. benefit aspect to the answer here.

Reply:
Yep, your analysis is spot on. This was the easiest to implement; happy to talk through the other options.

For background - the request body to FireFly for creating an ERC20 token pool looks something like:
```
{
  "name": "FlapCoin",
  "symbol": "FFT",
  "type": "fungible",
  "config": {
    "address": "0x1111111111111"
  }
}
```
Name and symbol are first-class FireFly things. Name must be unique within the namespace and conform to FireFly restrictions for names. This is a potential problem area for any solution here, as we can't really guarantee all deployed contracts will have names that are acceptable to FireFly as pool names. We could have two different names, but that's confusing. We could also ignore the name from the contract (which is what we do currently).

Symbol is optional and can be pretty much any string, so it's less problematic.